### DISCUSSION TITLE
class에서 return this를 함으로써 얻을 수 있는 기대 효과를 알려주세요.

### DISCUSSION THREAD

**Discussion by javascriptbiggosoo** [wanted-9th-3team/frontend-study#61] [프론트 기술 질문 공간]
많은 라이브러리 메소드에서 return이 this인 경우를 많이 볼 수 있습니다.
![image](https://github.com/wanted-9th-3team/frontend-study/assets/28939278/ff2c1797-a91b-46f2-a7fe-65b03f0b75ed)
사진: https://leafletjs.com/reference.html#popup

무슨 목적으로 return this를 많이 사용하는 것일까요?

**Comment by michoball** [comment]
메서드 체이닝을 사용해서 보다 유연한 프로그래밍을 사용할 수 있도록 하기 위함입니다. 

예를 들어 다음과 같은 클래스가 있다면 

```javascript
class 아무거나 {
  constructor() {
    this.value = 뭐든
   }

메서드1 ( ) { 
  작업을 하고 
   return this
 }

메서드 2 (num ) {
 다른 작업을 하고 
 return this
 }

값 반환 ( ) {
 return this.value
} 

}
```

이런 클래스가 있다면 메서드의 리턴값으로 this를 사용했기 때문에 
```javascript
const someClass = new 아무거나 ( )

const someResult = someClass.메서드1( ).메서드2( 5 ).값반환( )

console.log(someResult) //  value에 작업을하고 5를 가지고 다른 작업을 한 값이 출력
```
와 같은 작업을 할 수 있습니다.

**Reply by javascriptbiggosoo** [reply]
네 return this를 포함하는 메소드들은 보통 아무 값도(undefined) return 해주지 않아도 주어진 역할을 해낼 수 있는 경우가 대부분입니다.
하지만 return this가 없었다면 주어진 예시 코드
```jsx
const someResult = someClass.메서드1( ).메서드2( 5 ).값반환( )
```
는
```jsx
someClass.메서드1()
someClass.메서드1(5)
const someResult = someClass.값반환()
```
로 작성했어야 했겠죠.


return this 처리는 map, filter, reduce 시리즈가 메소드체이닝으로 array 가공을 수월하개 해준것처럼 코딩경험을 향상시킵니다.
